code/game/machinery/computer/HolodeckControl.dm

```diff
@@ -383,9 +383,27 @@
 /obj/structure/rack/holorack
 	can_deconstruct = FALSE
 
+
+/obj/structure/stool/holostool
+	can_deconstruct = FALSE
+
+
 /obj/item/holo
 	damtype = STAMINA
 
+/obj/item/stool/holostool
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/stool/holostool/attack(mob/M as mob, mob/user as mob)
+	if(prob(5) && istype(M,/mob/living))
+		user.visible_message("<span class='danger'>[user] breaks [src] over [M]'s back!.</span>")
+		user.unEquip(src)
+		qdel(src)
```

Comment:
Can you do this in a more OOP manner? Instead of having the bunch of code copy-pasted from its parent with the metal sheet removed, add a variable called metal_bash_amount (Or whatever clear concise name fancy you), make it so that it generates a stack of said amount, and set the variable to 0 on holostool.

Also, avoid a istype check by typecasting more strictly in the parameters of the variable

Reply:
I will try that. I've not had enough practice yet doing things that way yet. I'll try to get to it later in the week, maybe tomorrow or Saturday